tests/tests/classes/Plugin.js

```diff
@@ -0,0 +1,160 @@
+
+  it('should return a valid name', () => {
+    const name = SPlugin.getName();
+    expect(name).to.be.equal('Plugin');
+  });
+
+  describe('Add action', () => {
+
+    before((done) => {
+      SPlugin.addAction(actionMock, actionMockConfig);
+      done();
+    });
+
+    after((done) => {
+      done();
+    });
+
+    it('should have an options property when adding an action', () => {
+      expect(serverless.commands.action.mock).to.have.property('options');
+    });
+
+    it('should have an action property when adding an action', () => {
+      expect(serverless.actions).to.have.property('actionMock');
+    });
+
+    it('should have a mock property when adding an action', () => {
+      expect(serverless.commands.action).to.have.property('mock');
+    });
+
+    it('should have a parameters property when adding an action', () => {
+      expect(serverless.commands.action.mock).to.have.property('parameters');
+    });
+
+    it('should add a context action when adding an action', () => {
+      expect(serverless.commands.action.mock.contextAction).to.equal('mock');
+    });
+
+    it('should add a handler when adding an action', () => {
+      expect(serverless.commands.action.mock.handler).to.equal('actionMock');
+    });
+
+    it('should add a context when adding an action', () => {
+      expect(serverless.commands.action.mock.context).to.equal('action');
+    });
+
+    it('should have a specified options array when adding an action', () => {
+      expect(serverless.commands.action.mock.options.length).to.equal(1);
+    });
+
+    it('should have an option inside the options array when adding an action', () => {
+      expect(serverless.commands.action.mock.options[0].option).to.equal('mockOption');
+    });
+
+    it('should have a shortcut inside the options array when adding an action', () => {
+      expect(serverless.commands.action.mock.options[0].shortcut).to.equal('m');
+    });
+
+    it('should have a description inside the options array when adding an action', () => {
+      expect(serverless.commands.action.mock.options[0].description).to.equal('Mock option');
+    });
+
+    it('should have a specified parameters array when adding an action', () => {
+      expect(serverless.commands.action.mock.parameters.length).to.equal(1);
+    });
+
+    it('should have a parameter inside the parameters array when adding an action', () => {
+      expect(serverless.commands.action.mock.parameters[0].parameter).to.equal('mockParam');
+    });
+
+    it('should have a description inside the parameters array when adding an action', () => {
+      expect(serverless.commands.action.mock.parameters[0].description).to.equal('Mock param');
+    });
+
+    it('should have a position inside the parameters array when adding an action', () => {
+      expect(serverless.commands.action.mock.parameters[0].position).to.equal('0->');
+    });
+
+    // this test adds an additional hook
+    it('should runs the specified action', () => {
```

Comment:
My assumption is because we're hardcoding SPlugin for the whole class, but it should be recreated for each test so we're actually working with a new object every time.